### PR TITLE
A new option to set how a song is considered as played

### PR DESCRIPTION
**Pull request by jej**
Check-list
----------

 * [x] There is a linked issue discussing the motivations for this feature or bugfix
 * [ ] Unit tests have been added where possible
 * [x] I've added / updated documentation for any user-facing features.
 * [x] Performance seems to be comparable or better than current `master`

What this change is adding / fixing
-----------------------------------
See issue #3874

### CODE REVIEW

**Comment by jej** [comment]
Strange, the falling test runs on my side...
```
$ ./setup.py test --to-run=TSongTracker
running test
========================================================================================================= test session starts =========================================================================================================
platform linux -- Python 3.9.7, pytest-6.0.2, py-1.10.0, pluggy-0.13.0
rootdir: /usr/local/src/quodlibet, configfile: setup.cfg
plugins: anyio-3.4.0
collected 4520 items / 4515 deselected / 5 selected                                                                                                                                                                                   

tests/test_qltk_tracker.py .....                                                                                                                                                                                                [100%]

========================================================================================================== warnings summary ===========================================================================================================
tests/test_util_copool.py:15
  /usr/local/src/quodlibet/tests/test_util_copool.py:15: PytestUnknownMarkWarning: Unknown pytest.mark.flaky - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.flaky(max_runs=3, min_passes=2)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
============================================================================================ 5 passed, 4515 deselected, 1 warning in 6.63s ============================================================================================
```

File: quodlibet/config.py [hunk 1]
```diff
@@ -49,6 +49,7 @@
 
         "is_playing": "false",
         "restore_playing": "false",
+        "consider_played_percent": "50",
```
Comment (declension) [inline quodlibet/config.py:52]: Not sure about this name (_consider_ feels a bit weird).
```suggestion
        "playcount_minimum_length_proportion": "0.5",
```
or something along those lines?

Also, all (new) config items should have a comment to explain them

File: quodlibet/qltk/prefs.py [hunk 2]
```diff
@@ -433,13 +434,45 @@ def __init__(self):
...
+            scale.set_value_pos(Gtk.PositionType.LEFT)
+            scale.set_show_fill_level(True)
+            scale.set_tooltip_text(_("If a track is played beyond this percentage, "
+                             "its number of plays and last play date are updated."))
+            scale.connect('format-value',
+                            lambda _, value: "%d%%" % value)
+            scale.connect('value-changed', scale_changed)
+
+            consider_played_label = Gtk.Label(
+                label=_('Consider a track as played after listening'))
```
Comment (declension) [inline quodlibet/qltk/prefs.py:457]:
```suggestion
                label=_('Consider a track as played after listening to this proportion of its overall length'))
```

**Comment by declension** [comment]
Looks good generally, but I think this should be kept in the Advanced Preferences plugin as most users will never want to change it, but some will - and that's where we keep those config items generally
Thanks

**Comment by jej** [comment]
Thanks for your feedback.
It's ok for me concerning your suggestions. I can continue to work on these changes.

**Comment by jej** [comment]
What about this?

**Comment by jej** [comment]
By the way I added a new widget slider to the advanced pref plugin...

capture_ecran_20220110-114010

**Comment by bassstorm** [comment]
> By the way I added a new widget slider to the advanced pref plugin...

Marvellous!

File: quodlibet/qltk/prefs.py [hunk 3]
```diff
@@ -433,13 +434,45 @@ def __init__(self):
             self.pack_start(f, False, True, 0)
 
             vbox = Gtk.VBox()
+
```
[the remark this answers is not in the document]
Comment (jej) [reply quodlibet/qltk/prefs.py]: I can create a new clean branch if you prefer...

File: quodlibet/qltk/tracker.py [hunk 4]
```diff
@@ -153,7 +153,9 @@ def __end(self, player, song, ended, librarian, pl):
             else:
                 config.set("memory", "seek", 0)
 
-            if self.elapsed > 0.5 * song.get("~#length", 1):
+            if self.elapsed >= config.getint(
```
[the remark this answers is not in the document]
Comment (jej) [reply quodlibet/qltk/tracker.py]: Sorry, I didn't receive notification for your PR comments.
Ok, I will do it soon.

File: quodlibet/config.py (hunk 1 above)
Comment (jej) [reply quodlibet/config.py]: Done.

**Comment by jej** [comment]
@declension Should be ok now...

**Comment by jej** [comment]
Oups no sorry, doc needs a bit of update...

**Comment by declension** [comment]
Thanks!

**Comment by jej** [comment]
Cool, thanks too.
I will propose a new PR for #3881. It was conflicting because of the different changes in adv pref plugin.

**Pull request merged**